fineract-core/src/main/java/org/apache/fineract/commands/service/CommandSourceService.java

```diff
@@ -108,6 +110,16 @@ public ErrorInfo generateErrorInfo(Throwable t) {
         return errorHandler.handle(ErrorHandler.getMappable(t));
     }
 
+    @Transactional(propagation = Propagation.REQUIRES_NEW)
+    public void saveFailedCommandSourcesNewTransaction(final List<CommandSource> commandSources, final String errorMessage) {
+        commandSources.forEach(commandSource -> {
+            commandSource.setStatus(CommandProcessingResultType.ERROR);
```

Comment:
I am wondering on this...I am not a big fan of HTTP 500... neither to mark all of them as error... let me think it through...